etc/releasing.md

```diff
@@ -209,6 +208,49 @@ pushed.
   `docs/`. This may require pushing the commit to a fork of the C++ Driver
   repository and creating a pull request.
 
+## Update CHANGELOG.md for next release
+
+CHANGELOG.md on the `master` branch contains sections for every release. This is intended to ease searching for changes among all releases.
+CHANGELOG.md on a release branch (e.g. `releases/v1.2`) contains entries for all prior releases, and all patch releases on the `releases/v1.2` branch.
+
+### For a minor release (e.g. r1.2.0):
+
+Open a PR to the master branch to add a new empty `[Unreleased]` section to the CHANGELOG.md for the next minor release and next patch release. Example:
+
+```md
+## 1.3.0 [Unreleased]
+
+<!-- Will contain entries for 1.3.0 release. -->
+
+## 1.2.1 [Unreleased]
+
+<!-- Will contain entries for 1.2.1 release. Entries are implicitly included in 1.3.0 release. -->
+
+```
+
+Open a PR to the release branch (e.g. `releases/v1.2`) to add a new empty `[Unreleased]` section to the CHANGELOG.md for the next patch release. Example:
```

Comment:
This step should be covered by the "Branch if necessary" section below that comes _after_ the release process has completed. Additionally, given relocation of the "Update CHANGELOG.md for next release" instructions, the required subsequent edit will likely involve _removal_ of the minor release section stub.

I recommend deferring the addition of `[Unreleased]` stubs (for both master and release branches) to _after_ the "Branch if necessary" section, but before docker/packaging instructions:

- Audit JIRA tickets.
- Remove `[Unreleased]` for current release _and commit_ (important).
- Create release tag.
- Run "make release" scripts.
- Add `[Unreleased]` stubs for next release(s) _and commit_ (important).
- Create a new release branch at this point if necessary.
- Push new/updated release branch to remote (note: PR not required!).
- Create PR for all post-release changes targeting `master`.

This allows for the release tag to be correctly timed _after_ removal of `[Unreleased]` in the CHANGELOG, remain stable post-release (no post-PR-merge recommit shenanigans), and cleanly workaround `master` branch protection by pushing the post-release commits to the _release_ branch first instead.

Preferably the post-release PR to `master` should use "Rebase and merge" to preserve the individual commits (a la cherry-picking), but a "Squash and merge" is also fine. Additionally, to avoid mixing/merging up the release branch history with `master`, the "Create PR for post-release rX.Y -> master" can be preceeded by a new (temporary, to-be-deleted post-PR) branch separate from `releases/vX.Y`.

Reply:
Updated changes include a `Update CHANGELOG.md post-release ...` after the `Branch if necessary` section with use of a `post-release-changes` branch.